Microsoft.Toolkit.Uwp.UI.Controls/TabView/TabView.cs

```diff
@@ -0,0 +1,395 @@
+                    }
+
+                    if (available > TabWidthProvider.MinimumWidth)
+                    {
+                        // Constrain the column based on our required and available space
+                        tabc.MaxWidth = available;
+                    }
+
+                    //// TODO: If it's less, should we move the selected tab to only be the one shown by default?
+
+                    if (available <= TabWidthProvider.MinimumWidth)
+                    {
+                        tabc.Width = new GridLength(TabWidthProvider.MinimumWidth);
+                    }
+                    else if (required >= available)
+                    {
+                        // Fix size as we don't have enough space for all the tabs.
+                        tabc.Width = new GridLength(tabc.MaxWidth);
+                    }
+                    else
+                    {
+                        // We haven't filled up our space, so we want to expand to take as much as needed.
+                        tabc.Width = GridLength.Auto;
+                    }
+                }
+            }
+        }
+
+        private void AddTabButton_Click(object sender, RoutedEventArgs e)
+        {
+            AddTabButtonClick?.Invoke(this, new EventArgs());
+        }
+
+        private void TabView_SelectionChanged(object sender, SelectionChangedEventArgs e)
+        {
+            if (SelectedItem == null)
+            {
+                _tabContentPresenter.Content = null;
+                _tabContentPresenter.ContentTemplate = null;
+            }
+            else
+            {
+                var container = ContainerFromItem(SelectedItem) as TabViewItem;
+                if (container != null)
+                {
+                    _tabContentPresenter.Content = container.Content;
+                    _tabContentPresenter.ContentTemplate = container.ContentTemplate;
+                }
+            }
+
+            // If our width can be effected by the selection, need to run algorithm.
+            if (TabWidthProvider != null && TabWidthProvider.IsSelectedTabWidthDifferent)
+            {
+                TabView_SizeChanged(sender, null);
+            }
+        }
+
+        /// <inheritdoc/>
+        protected override void PrepareContainerForItemOverride(DependencyObject element, object item)
+        {
+            var tvi = element as TabViewItem;
+
+            tvi.Loaded += TabViewItem_Loaded;
+
+            if (tvi.Header == null)
+            {
+                tvi.Header = item;
+            }
+
+            if (tvi.HeaderTemplate == null)
+            {
+                tvi.HeaderTemplate = ItemHeaderTemplate;
+            }
+
+            if (tvi.IsClosable == null)
+            {
+                tvi.IsClosable = CanCloseTabs;
+            }
+
+            base.PrepareContainerForItemOverride(element, item);
+        }
+
+        private void TabViewItem_Loaded(object sender, RoutedEventArgs e)
```

Comment:
Make a Closing event of the TabViewItem that fires when the close button is clicked. Remove this event handler and subscribe to Closing in the PrepareContainerForItemOverride method